### PR TITLE
[tech] use crate chrono tz for timezone

### PR DESCRIPTION
**Pull request by patochectp**
ref: ND-940

### CODE REVIEW

File: src/objects.rs [hunk 1]
```diff
@@ -1809,6 +1812,26 @@ impl AddPrefix for GridRelCalendarLine {
...
+    }
+}
+
+impl FromStr for TzExt {
+    type Err = String;
+    fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
+        match Tz::from_str(s) {
+            Ok(timezone) => Ok(TzExt(timezone)),
+            Err(err) => Err(err),
+        }
```
Comment (datanel) [inline src/objects.rs:1831]:
```suggestion
        Tz::from_str(s).map(|tz| TzExt(tz))
```

File: src/objects.rs [hunk 2]
```diff
@@ -344,8 +345,9 @@ pub struct Network {
     pub url: Option<String>,
     #[serde(skip)]
     pub codes: KeysValues,
+    // #[derivative(Default(value = "\"Europe/Paris\".parse().ok()"))]
```
Comment (datanel) [inline src/objects.rs:348]: you can remove this comment.

File: src/objects.rs [hunk 3]
```diff
@@ -1809,6 +1812,26 @@ impl AddPrefix for GridRelCalendarLine {
     }
 }
 
+#[derive(Debug, Serialize, Deserialize, Clone, Copy, PartialEq)]
+pub struct TzExt(pub Tz);
```
Comment (pbougue) [inline src/objects.rs:1816]: It's worth commenting why you had to do that, and maybe rename:
```suggestion
pub struct ParsableTz(pub Tz);
```

Comment (patochectp) [reply src/objects.rs:1816]: we may have other needs so I prefer a generic name

**Pull request merged**